sweepai/utils/graph.py

```diff
@@ -182,6 +182,19 @@ def extract_first_degree(self, file_path: str) -> str:
             res += format_path(path, separator=" used in ") + "\n"
         return res
 
+    def topological_sort(self, file_paths: list[str]):
+        # Create a copy of the graph
+        graph = self.definitions_graph.copy()
+
+        # Filter out the nodes that are not in the file_paths
+        graph.remove_nodes_from([node for node in graph if node not in file_paths])
```

Comment:
we should filter out if its not in the file paths or not in the neighbors of the file paths (this is a bipartite graph)